src/bp/nlu-core/tools/math.ts

```diff
@@ -17,7 +13,15 @@ export function euclideanDistance(a: number[], b: number[]): number {
     const diff = b[i] - a[i]
     total += diff * diff
   }
-  return Math.sqrt(total)
+  return total
+}
+
+/**
+ * Vectorial distance between two N-dimentional points
+ * a[] and b[] must be of same dimention
+ */
+export function euclideanDistance(a: number[], b: number[]): number {
```

Comment:
can you add on your little todo to get rid of those funcs ... they all exist in the math.js library and they're probably much more optimized than those.

Reply:
sure